File: python/66c_1430B_Barrels.py
def solve(arr, n, k):
    arr.sort()
    for i in range(1, k+1):
        arr[n-1] += arr[n-1-i]
    return arr[n-1] - 0

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n, k = map(int, input().split())
        arr = list(map(int, input().split()))
        ans.append(solve(arr, n, k))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/66c_1430B_Barrels.cpp
#include<bits/stdc++.h>
typedef long long ll;
using namespace std;

ll solve(vector<ll> arr, int n, int k){
    sort(arr.begin(), arr.end());
    for(int i = 1; i <= k; i++){
        arr[n-1] += arr[n-1-i];
        arr[n-1-i] = 0;
    }
    return arr[n-1]-0;
}

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    int t, n, k;
    ll temp;
    vector<ll> ans;
    cin>>t;

    for(int i = 0; i < t; i++){
        cin>>n>>k;
        vector<ll> arr;
        for(int j = 0; j < n; j++){
            cin>>temp;
            arr.push_back(temp);
        }
        ans.push_back(solve(arr, n, k));
    }

    for(auto i : ans){
        cout<<i<<"\n";
    }
    return 0;
}